### PR TITLE
docs: fix: general rfc fixes

### PR DESCRIPTION
**Pull request by esavier**
general cleanup:
* template
* formatting
* paragraph titles
* other random stuff
* archive

### CODE REVIEW

**Review by frondeus — DISMISSED** [review]

**Review by frondeus — APPROVED** [review]

File: docs/rfc/0012_CDL_documentation_generation_01.md [hunk 1]
```diff
@@ -62,13 +60,11 @@ In general - it requires some significant work from team to switch fully to conf
 
 ## Generate configs from rust
```
Comment (smores56) [inline docs/rfc/0012_CDL_documentation_generation_01.md:61]:
```suggestion
## Generate Configs From Rust
```

File: docs/rfc/0012_CDL_documentation_generation_01.md [hunk 2]
```diff
@@ -62,13 +60,11 @@ In general - it requires some significant work from team to switch fully to conf
...
 
-### Compiler pass way:
-
+### Compiler pass way
```
Comment (smores56) [inline docs/rfc/0012_CDL_documentation_generation_01.md:63]:
```suggestion
### Compiler Pass Approach
```

File: docs/rfc/0012_CDL_documentation_generation_01.md [hunk 3]
```diff
@@ -62,13 +60,11 @@ In general - it requires some significant work from team to switch fully to conf
...
 We can create a binary that walks through our codebase and based on some criteria generates specs based on rust AST. It's doable, although not an easy task.
 More can be read searching for feature `rustc_private`, it allows users to use rustc as a library.
 
-### XTask + Lib way:
-
+### XTask + Lib way
```
Comment (smores56) [inline docs/rfc/0012_CDL_documentation_generation_01.md:67]:
```suggestion
### XTask + Lib Approach
```

File: docs/rfc/0012_CDL_documentation_generation_01.md [hunk 4]
```diff
@@ -89,17 +85,14 @@ and I'm not sure how much of an effort would it be to replace generated files wi
 Some context may be found in [this issue][tarpc-rs-161], it appears it's possible. For `xtask` this still would mean custom `.protobuf` gen.
 
 ## Mixed solution
```
Comment (smores56) [inline docs/rfc/0012_CDL_documentation_generation_01.md:87]:
```suggestion
## Mixed Solution
```

File: docs/rfc/0012_CDL_documentation_generation_01.md [hunk 5]
```diff
@@ -89,17 +85,14 @@ and I'm not sure how much of an effort would it be to replace generated files wi
...
-
 It's not a problem to have mixed solution in some places. Important part is nothing has to be done manually in more than one place.
 
 # Author's comment
```
Comment (smores56) [inline docs/rfc/0012_CDL_documentation_generation_01.md:90]:
```suggestion
# Author's Comment
```

File: docs/rfc/0000_Template_00.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+* # Please Use Title Case
+-->
+
+## Glossary
+
+##### Terminology
+
+##### Features
+
+#####  References
```
Comment (smores56) [inline docs/rfc/0000_Template_00.md:50]: No need for a References section header, as the references won't be visible in the rendered document.

File: docs/rfc/0003_Usage_of_Message_Pack_format_as_a_CDL_input_01.md [hunk 7]
```diff
@@ -46,7 +41,7 @@ v1 Input Format - messages of format v1 are not related to Message Batching, i.e
 	    data: { any valid json },
 	}
 ```
-v1 Batch - it is a batch format introduced to alleviate some problems with transports. It conforms only to the v1 input format, and its treatment is described in RFC for [CDLF-00009-00][2]
+v1 Batch - it is a batch format introduced to alleviate some problems with transports. It conforms only to the v1 input format, and its treatment is described in RFC for batching
```
Comment (smores56) [inline docs/rfc/0003_Usage_of_Message_Pack_format_as_a_CDL_input_01.md:44]:
```suggestion
v1 Batch - it is a batch format introduced to alleviate some problems with transports. It conforms only to the v1 input format, and its treatment is described in the batching RFC.
```

File: docs/rfc/0003_Usage_of_Message_Pack_format_as_a_CDL_input_01.md [hunk 8]
```diff
@@ -130,20 +125,20 @@ Both formats can use this same, abstract, transport endpoint. All the informatio
...
 * Each transport have to get additional code, handling different ways of getting, and possibly parsing, the metadata provided with the message.
 * It was also stated on the internal meetings that clients and/or libraries in different languages may have problems with this functionality, as it is not widely used. As per project directives, CDL have to be compatible with different languages, and this may be a potential issue.
 
-##### Performance:
+##### Performance
 * Assuming the difference in metadata formats and our ability to both parse and serialize it, it can be safely assumed that performance will not suffer, however it depends on the way each library will get and use the metadata provided with the message. Comparing to the usual `O(m*m℘)` where `m` is the message size and `m℘` is the parsing cost, this solution will cost roughly `O((m*m℘) + (n*n℘))` where `n` and `n℘` is metadata length and its parsing cost respectively.
 
-##### Solution III - single endpoint with format recognition
+##### Solution III - single Endpoint With Format Recognition
```
Comment (smores56) [inline docs/rfc/0003_Usage_of_Message_Pack_format_as_a_CDL_input_01.md:141]:
```suggestion
##### Solution III - Single Endpoint With Format Recognition
```

File: docs/rfc/0003_Usage_of_Message_Pack_format_as_a_CDL_input_01.md [hunk 9]
```diff
@@ -225,48 +220,38 @@ This feature MUST undergo thorough testing before being accepted. Test cases MUS
...
 #### Notes
 * CDLF-00009-00 - Message Batching - have to be taken into consideration while writing this feature
 * CDLF-0000A-00 - Message Ordering - guaranteeing order in case of mixed formats will not be possible, albeit while using a specific format or different endpoint, order can be enforced.
 
 ### End Matter
-
 This is the version of the RFC that was deliberately cut down to ingestible size and format. The number of cases and solutions were condensed to those that were considered by the researcher the most valuable, leaving out minor variation that can be deduced from already provided solutions.
 
-#####  References:
+#####  References
```
Comment (smores56) [inline docs/rfc/0003_Usage_of_Message_Pack_format_as_a_CDL_input_01.md:243]: Ditto on the references section header being unnecessary.

**Review by smores56 — CHANGES_REQUESTED** [review]

**Pull request merged**